low-level/framework/messaging/thread_controller.py

```diff
@@ -193,12 +317,11 @@ def run(self):
                 thread_init_status = m.get_thread_init_status()
                 logger.debug("Thread status for {} is {}".format(
                     m.__class__, thread_init_status))
-                if thread_init_status == SensorThreadState.FAILED:
-                    m.shutdown()
-                elif thread_init_status == SensorThreadState.WAITING:
+                if thread_init_status == SensorThreadState.WAITING:
```

Comment:
Here I think it should be => `if thread_init_status == SensorThreadState.FAILED:` 
with the latest change, if initialize() is successful we are keeping the status SUCCESS, if any exception status is FAILED, so WAITING status is never assigned, it is getting overwritten. so ` continue_waiting` will never be set to True. and we will send the success msg anyway.

Reply:
Yes, right. I will explain it. As per the requirement(EOS-22452), `allowing other sensors to continue, thus keeping the monitoring impact minimal`.
During initialize() or run() time failure, status won't be set to `WAITING` because we shutdown that sensor for recovery.
The reasons why avoid continue_waiting for FAILED case are
   1. If any sensor is failed, thread status will be RUNNING until recovery cycle time.
   2. If not recoverable, fault alert will be sent and sensor thread will be terminated.

@sumedhak27

Reply:
Corrected, @sumedhak27 
Please review, module_thread.py

Reply:
`allowing other sensors to continue, thus keeping the monitoring impact minimal`
 Agreed 👍 
`While starting a thread(before initialize), thread state is set to WAITING until it completes.
During initialize() or run() time failure, init status will be set to FAILED and recovery process starts`
True, even if one of the sensor threads is down we are sending sspl success msg on which sspl-test is waiting.  
File =>`sspl-test/sspl_start_checker`
so now if we want to send this msg anyway either somehow we need to disable tests for those sensors or send the msg only when all the sensor threads are up and running.
OR those tests will simply fail right.. that's what I wanted to put.